dftimewolf/lib/processors/gce_forensics_vm.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+"""Creates an analysis VM and attaches GCP disks to it for analysis."""
+
+from typing import List, Optional, Dict
+
+from libcloudforensics import errors as lcf_errors
+from libcloudforensics.providers.gcp import forensics as gcp_forensics
+from libcloudforensics.providers.gcp.internal import common
+from libcloudforensics.providers.gcp.internal import project as gcp_project
+from libcloudforensics.providers.gcp.internal import compute
+
+from dftimewolf.lib import module
+from dftimewolf.lib.containers import containers
+from dftimewolf.lib.modules import manager as modules_manager
+from dftimewolf.lib.state import DFTimewolfState
+
+
+class GCEForensicsVM(module.BaseModule):
+  """Google Cloud Forensics VM creator.
+
+  Attributes:
+    project (gcp.GoogleCloudProject): Project in which to create the VM.
+    analysis_vm (gcp.GoogleComputeInstance): Analysis VM on
+         which the disk copy will be attached.
+    incident_id (str): Incident identifier used to name the analysis VM.
+    boot_disk_size (Optional[float]): Optional. Size of the analysis VM boot
+        disk (in GB).
+    boot_disk_type (Optional[str]): Optional. Disk type to use.
+    cpu_cores (Optional[int]): Optional. Number of CPU cores to create the VM
+        with.
+    image_project (Optional[str]): Optional. Name of the project where the
+        analysis VM image is hosted.
+    image_family (Optional[str]): Optional. Name of the image to use to
+        create the analysis VM.
+  """
+
+  _ANALYSIS_VM_CONTAINER_ATTRIBUTE_NAME = 'Analysis VM'
+  _ANALYSIS_VM_CONTAINER_ATTRIBUTE_TYPE = 'text'
+
+  def __init__(self,
+               state: DFTimewolfState,
+               name: Optional[str]=None,
+               critical: bool=False) -> None:
+    """Initializes a Google Cloud Platform (GCP) collector.
+
+    Args:
+      state (DFTimewolfState): recipe state.
+      name (Optional[str]): The module's runtime name.
+      critical (Optional[bool]): True if the module is critical, which causes
+          the entire recipe to fail if the module encounters an error.
+    """
+    super(GCEForensicsVM, self).__init__(
+        state, name=name, critical=critical)
+    self.project = None  # type: gcp_project.GoogleCloudProject
+    self.analysis_vm = None  # type: compute.GoogleComputeInstance
+    self.incident_id = str()
+    self.boot_disk_size = 0.0
+    self.boot_disk_type = str()
+    self.cpu_cores = 0
+    self.image_project = str()
+    self.image_family = str()
+    self._gcp_label = {}  # type: Dict[str, str]
+
+  # pylint: disable=arguments-differ,too-many-arguments
+  def SetUp(self,
+            project_name: str,
+            incident_id: Optional[str]=None,
+            zone: str='us-central1-f',
+            boot_disk_size: float=50,
+            boot_disk_type: str='pd-standard',
+            cpu_cores: int=4,
+            image_project: str='ubuntu-os-cloud',
+            image_family: str='ubuntu-1804-lts') -> None:
```

Comment:
`--no-analysis-vm`: Yeah the original module had that, where it does both tasks, disk copying and vm launching, and there it made sense. Splitting the functionality out makes it seem silly to have a flag "actually, don't run this whole module in this recipe." If we wanted to optionally skip modules in a recipe it would make more sense to have something at the orchestration layer, but that's a whole different effort. 

I'll add it in here for consistency with the module it's replacing, but now that I've written it down, I think orchestration improvements, that allow for optional recipes might go far to reducing recipe count, where recipes have one module difference. Will raise a separate bug for that.

Reply:
Done, + test

Reply:
Yeah I agree, tweaking the orchestration layer to account for optional modules would be a good idea. Thanks for opening an issue.